### ISSUE TITLE
Deploying local ovf results in 'Error: error while importing OVF template error while uploading the file file.vmdk": dial tcp: lookup *: no such host' #1096

### ISSUE THREAD

**Issue by pawelprazak** [hashicorp/terraform-provider-vsphere#1806]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

any

### Terraform Provider

current

### VMware vSphere

6.0

### Description

Issue https://github.com/hashicorp/terraform-provider-vsphere/issues/1096 is still valid and should be re-opened, 
because deployment of an OVF/OVA is possible with `govc` and ESXi, without vCenter server.

There is an unnecessary validation in the terrraform provider, the XML is correct to have `*`, because I've validated this with the SOAP XML dumps form the `govc` successful run.

### Affected Resources or Data Sources

vsphere_virtual_machine

### Terraform Configuration

irrelevant

### Debug Output

irrelevant

### Expected Behavior

To be able to deploy OVF/OVA to an ESXi instance, like it is possible with `govc`.

### Actual Behavior

```
error while importing ovf/ova template, error while uploading the disk ubuntu-focal-20.04-cloudimg.vmdk error while uploading the file ./focal-server-cloudimg-amd64/ubuntu-focal-20.04-cloudimg.vmdk Post "https://*/ha-nfc/529414ed-46c2-3e90-89e1-7bc4828545c9/disk-0.vmdk": dial tcp: lookup *: no such host
```

### Steps to Reproduce

Deploy an OVF or OVA to an ESXi provider and the error appears.

### References

#1096
#1105 
#1310 
#1235

**Comment by tenthirtyam** [comment]
Please test on ESXi 7.0 or later and update the below. vSphere 6.x is no longer under general support.

> ### VMware vSphere
> 6.0

**Comment by pawelprazak** [comment]
> Please test on ESXi 7.0 or later and update the below. vSphere 6.x is no longer under general support.
> 
> > ### VMware vSphere
> > 6.0

I don't have an ability to do so, but it won't change anything, because `govc` works as expected and the provider has unnecessary validation of the URL, that will contain `*` in the SOAP XML and it is expected.

**Comment by pawelprazak** [comment]
TL;DR: this logic seems to be missing in the provider and is likely the root cause of this issue:
```
	host, _ := splitHostPort(u.Host)
	if host == "*" {
		// Also use Client's port, to support port forwarding
		u.Host = c.URL().Host
	}
```

**Comment by tenthirtyam** [comment]
Please feel free to submit a pull request with the fix and test results.

**Comment by pawelprazak** [comment]
> Please feel free to submit a pull request with the fix and test results.

I'd love to, but it might require some refactoring, instead of just adding this logic on top of the current state.

The best approach IMHO would be to reuse this function:
https://github.com/vmware/govmomi/blob/939e78dc19bf32a5fef077079e9263e058ea5ee3/vim25/soap/client.go#L704

But it would have to be exposed somehow and be made available to the:
https://github.com/hashicorp/terraform-provider-vsphere/blob/24846008ab2a9fa630c4e41081809b81b71f912b/vsphere/internal/helper/ovfdeploy/ovf_helper.go#L125

Please advice on how would you prefer to accomplish that.

One way would be to pass this reference:
https://github.com/hashicorp/terraform-provider-vsphere/blob/24846008ab2a9fa630c4e41081809b81b71f912b/vsphere/resource_vsphere_virtual_machine.go#L1352
into this call:
https://github.com/hashicorp/terraform-provider-vsphere/blob/24846008ab2a9fa630c4e41081809b81b71f912b/vsphere/resource_vsphere_virtual_machine.go#L1367
and pass it all the way down to the
https://github.com/hashicorp/terraform-provider-vsphere/blob/24846008ab2a9fa630c4e41081809b81b71f912b/vsphere/internal/helper/ovfdeploy/ovf_helper.go#L125

Not very elegant but should work.

Also a pointer as to where and how to add a meaningful test would be appreciated :)
